### PR TITLE
Mangle columns in KT join

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

**Comment by tpoterba** [comment]
Depends on #2111

**Comment by tpoterba** [comment]
Will reopen after the first goes in

File: src/main/scala/is/hail/keytable/KeyTable.scala [hunk 1]
```diff
@@ -407,15 +407,17 @@ case class KeyTable(hc: HailContext, rdd: RDD[Row],
            |  Left signature: ${ TStruct(keyFields).toPrettyString(compact = true) }
            |  Right signature: ${ TStruct(other.keyFields).toPrettyString(compact = true) }""".stripMargin)
 
-    val overlappingFields = fieldNames.toSet.intersect(other.fieldNames.toSet) -- key -- other.key
-    if (overlappingFields.nonEmpty)
-      fatal(
-        s"""Columns that are not keys cannot be present in both key tables.
-           |  Overlapping fields: ${ overlappingFields.mkString(", ") }""".stripMargin)
+    val preNames = fieldNames ++ other.valueSignature.fields.map(_.name)
+    val (joinNames, remapped) = mangle(preNames)
```
Comment (danking) [inline src/main/scala/is/hail/keytable/KeyTable.scala]: I found the name `joinNames` a bit confusing, these are the de-duplicated names, right?

Comment (tpoterba) [reply src/main/scala/is/hail/keytable/KeyTable.scala]: yeah, I'll rename to `finalColumnNames` or something.

**Comment by tpoterba** [comment]
addressed comments, back to you.

**Pull request merged**